pkg/yurthub/cachemanager/storage_wrapper_test.go

```diff
@@ -0,0 +1,154 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cachemanager
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/openyurtio/openyurt/pkg/yurthub/storage"
+	"github.com/openyurtio/openyurt/pkg/yurthub/storage/disk"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func clearDir(dir string) error {
+	return os.RemoveAll(dir)
+}
+
+func TestStorageWrapper(t *testing.T) {
+	dir := fmt.Sprintf("%s-%d", rootDir, time.Now().Unix())
+	pod := runtime.Object(&v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "mypod1",
+			Namespace:       "default",
+			ResourceVersion: "1",
+		},
+	})
+
+	defer clearDir(dir)
+
+	dStorage, err := disk.NewDiskStorage(dir)
+	if err != nil {
+		t.Errorf("failed to create disk storage, %v", err)
+	}
+	sWrapper := NewStorageWrapper(dStorage)
+
+	t.Run("Test create storage", func(t *testing.T) {
+		err = sWrapper.Create("kubelet/pods/default/mypod1", pod)
+		if err != nil {
+			t.Errorf("failed to create obj, %v", err)
+		}
+		obj, err := sWrapper.Get("kubelet/pods/default/mypod1")
+		if err != nil {
+			t.Errorf("failed to create obj, %v", err)
+		}
+		accessor := meta.NewAccessor()
+		name, _ := accessor.Name(obj)
+		if name != "mypod1" {
+			t.Errorf("the name is not expected, expect mypod1, get %s", name)
+		}
+	})
+
+	t.Run("Test update storage", func(t *testing.T) {
+		updatePod := runtime.Object(&v1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Pod",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            "mypod1",
+				Namespace:       "default",
+				ResourceVersion: "1",
+				Labels: map[string]string{
+					"tag": "test",
+				},
+			},
+		})
+		err = sWrapper.Update("kubelet/pods/default/mypod1", updatePod)
+		if err != nil {
+			t.Errorf("failed to update obj, %v", err)
+		}
+		obj, err := sWrapper.Get("kubelet/pods/default/mypod1")
+		if err != nil {
+			t.Errorf("unexpected error occurs, %v", err)
+		}
+		accessor := meta.NewAccessor()
+		labels, _ := accessor.Labels(obj)
+		if vaule, ok := labels["tag"]; ok {
+			if vaule != "test" {
+				t.Errorf("failed to get label, expect test, get %s", vaule)
+			}
+		} else {
+			t.Errorf("unexpected error occurs, the label `tag` is not existed")
+		}
+
+	})
+
+	t.Run("Test list obj and keys", func(t *testing.T) {
+		_, err = sWrapper.ListKeys("kubelet/pods/default")
+		if err != nil {
+			t.Errorf("failed to list keys, %v", err)
+		}
+		_, err = sWrapper.List("kubelet/pods/default")
+		if err != nil {
```

Comment:
For a List request, if you get kubelet/pods, and there are no pods in the cache, it means that all pods may have been accidentally deleted. At this time, the `storageWrapper.List` function will return `storage.ErrStorageNotFound`. Maybe the unit test here needs to add this use case.
The specific logic code is here: https://github.com/openyurtio/openyurt/blob/146a74498f51781f1ec1f9a98055cf3fa4c02f87/pkg/yurthub/cachemanager/storage_wrapper.go#L161

Reply:
done

Reply:
Good idea. I have moved the case to last. Please review.

Reply:
Should we keep the use case of normal List requests for non-empty resources?

Reply:
Done.